website/docs/attached-class.md

```diff
@@ -0,0 +1,75 @@
+---
+id: attached-class
+title: T.attached_class
+---
+
+`T.attached_class` can be used to refer to the type of instances from a
+singleton-context, in a way that is friendly to inheritance.
+
+```ruby
+class AbstractData
+  extend T::Sig
+
+  sig {params(id: Integer).returns(T.attached_class)}
+  def self.load(id)
```

Comment:
I feel kind of bad using `load` in this example because technically `load` already exists on Kernel which makes this an incompatible override (which isn't super bad, but maybe not something we want to encourage with the flagship example).